src/main/java/uk/gov/hmcts/darts/transcriptions/service/impl/TranscriptionDownloader.java

```diff
@@ -44,37 +40,41 @@ public DownloadTranscriptResponse downloadTranscript(Integer transcriptionId) {
             .max(comparing(TranscriptionDocumentEntity::getUploadedDateTime))
             .orElseThrow(() -> new DartsApiException(FAILED_TO_DOWNLOAD_TRANSCRIPT));
 
-        var latestExternalObjectDirectory = latestTranscriptionDocument.getExternalObjectDirectoryEntities()
-            .stream()
-            .filter(dir -> nonNull(dir.getExternalLocation()))
-            .max(comparing(ExternalObjectDirectoryEntity::getCreatedDateTime))
-            .orElseThrow(() -> new DartsApiException(FAILED_TO_DOWNLOAD_TRANSCRIPT));
-
         auditApi.recordAudit(DOWNLOAD_TRANSCRIPTION, userAccountEntity, transcriptionEntity.getCourtCase());
 
         return DownloadTranscriptResponse.builder()
-            .resource(getResourceStreamFor(latestExternalObjectDirectory))
+            .resource(getResourceStreamFor(latestTranscriptionDocument))
             .contentType(latestTranscriptionDocument.getFileType())
             .fileName(latestTranscriptionDocument.getFileName())
-            .externalLocation(latestExternalObjectDirectory.getExternalLocation())
             .transcriptionDocumentId(latestTranscriptionDocument.getId()).build();
     }
 
     private UserAccountEntity getUserAccount() {
         return userIdentity.getUserAccount();
     }
 
-    private InputStreamResource getResourceStreamFor(ExternalObjectDirectoryEntity externalObjectDirectoryEntity) {
-        final UUID externalLocation = externalObjectDirectoryEntity.getExternalLocation();
-        InputStreamResource stream;
-        if (externalObjectDirectoryEntity.isForLocationType(UNSTRUCTURED)) {
-            stream = new InputStreamResource(dataManagementApi.getBlobDataFromUnstructuredContainer(externalLocation).toStream());
-        } else if (externalObjectDirectoryEntity.isForLocationType(INBOUND)) {
-            stream = new InputStreamResource(dataManagementApi.getBlobDataFromInboundContainer(externalLocation).toStream());
-        } else {
+    private InputStreamResource getResourceStreamFor(TranscriptionDocumentEntity latestTranscriptionDocument) {
+        var downloadableExternalObjectDirectories = DownloadableExternalObjectDirectories.getFileBasedDownload(
+            latestTranscriptionDocument.getExternalObjectDirectoryEntities());
+
+        dataManagementFacade.getDataFromUnstructuredArmAndDetsBlobs(downloadableExternalObjectDirectories);
+
+        var downloadResponseMetaData = downloadableExternalObjectDirectories.getResponse();
+
+        try {
+            if (!downloadResponseMetaData.isSuccessfulDownload()) {
```

Comment:
As well as this I would suggest adding the line to FileBasedDownloadResponseMetaData so if anything closes the input stream the file also gets cleaned up (this was something I originally missed):-

inputStream = new FileInputstream(fileToBeDownloadedTo.toURI()) {
                @Override
                public void close() {
                    FileBasedDownloadResponseMetaData.this.close();
                }

Reply:
I've left this for now as @hemantasharma1129 says that the file cleanup can be done in another ticket.
So as not to hold up @johnrace0 too much with his ticket